proofread: how_to_use_on_kubernetes.md (#1935)
juicedata/juicefs: docs/en/deployment/how_to_use_on_kubernetes.md

---
# Use JuiceFS on Kubernetes

JuiceFS is ideal for use as a storage layer for Kubernetes clusters, and currently has two common uses.

## JuiceFS CSI Driver

> [JuiceFS CSI Driver](https://github.com/juicedata/juicefs-csi-driver) follows the [CSI](https://github.com/container-storage-interface/spec/blob/master/spec.md) specification, implements the interface between the container orchestration system and the JuiceFS file system, and supports dynamic configured JuiceFS volumes for use by Pod.

Reviewer: 好像原文最后一句话强调的是“支持「动态配置JuiceFS卷」这一个功能”，是不是之前的翻译“supports the dynamic configuration of JuiceFS volumes for use by Pod” 更符合原文意思呢？

Author: emm，我是觉得 configuration 有点奇怪。这里应该是个名词，而不是形容词，不然不太符合原意。
static configured volume 应该指的是 kubernetes 内部直接创建的 PersistentVolume，dynamic configured volumes 应该指的是使用 PersistentVolumeClaim 在 Pod 创建时动态申请的 volume

#### Install with Helm

Helm is the package manager for Kubernetes, and Charts are packages managed by Helm. You can think of it as the equivalent of Homebrew formula, APT dpkg, or YUM RPM in Kubernetes.

Installing with Helm requires Helm **3.1.0** or higher. To learn more about how to install Helm, please refer to ["Helm Installation Guide"](https://github.com/helm/helm#install).

> 1. Prepare a configuration file for setting the basic information of the storage class, for example `values.yaml`. Complete the following configuration information and paste it into `values.yaml`. Specifically, the `backend` field is JuiceFS file system related information, you can refer to [JuiceFS Quick Start Guide](../getting-started/for_local.md) for more details. If you are using a JuiceFS volume that has been created in advance, you only need to fill the two fields `name` and `metaurl`. The `mountPod` part can be used to set the resource allocations of CPU and memory for the Pod using this driver. Fields not in use can be deleted, or you can leave their values blank.

Reviewer: ```suggestion
1. Prepare a configuration file for setting the basic information of the storage class, for example `values.yaml`. Complete the following configuration information and paste it into `values.yaml`. Specifically, the `backend` field is JuiceFS file system related information, and you can refer to [JuiceFS Quick Start Guide](../getting-started/for_local.md) for more details. If you are using a JuiceFS volume that has been created in advance, you only need to fill the two fields `name` and `metaurl`. The `mountPod` part can be used to set the resource allocations of CPU and memory for the Pod using this driver. Fields not in use can be deleted, or you can leave their values blank.
```

#### Install with kubectl

> Since Kubernetes deprecates some old APIs during the version bumping process, you need to select the applicable deployment file fit for the version of Kubernetes you are using:

Reviewer: ```suggestion
Since Kubernetes deprecates some old APIs during the version bumping process, you need to select the applicable deployment file that works for the version of Kubernetes you are using:
```

  csi.storage.k8s.io/provisioner-secret-namespace: kube-system
```

> Then deploy the storage class by `kubectl apply`:

Reviewer: ```suggestion
Then deploy the storage class with `kubectl apply`:
```

```shell
$ kubectl apply -f ./juicefs-sc.yaml
```

### Use JuiceFS as PersistentVolume

> JuiceFS CSI Driver supports both static and dynamic PersistentVolume. You can either manually assign the PersistentVolume previously created to Pods, or dynamically create volumes by using PersistentVolumeClaim when Pods are deployed.

Reviewer: ```suggestion
JuiceFS CSI Driver supports both static and dynamic PersistentVolume. You can either manually assign the PersistentVolume created previously to Pods, or dynamically create volumes by using PersistentVolumeClaim when Pods are deployed.
```

### Monitoring

> Please refer to ["Monitoring"](../administration/monitoring.md) to learn how to collect and show JuiceFS monitoring metrics.

Reviewer: Would "display" be better here?

Author: display 在计算机领域里感觉更像和「图像显示」本身相关的东西，我猜这里是想说如何呈现这些 metrics，比如通过表格还是折线图还是饼图等等。我不是很确定哈。

            privileged: true
```

> > ⚠️ **Risk Warning**: With the privileged mode being enabled by `privileged: true`, the container has access to all devices of the host, that is, it has full control of the host's kernel. Improper uses will bring serious safety hazards. Please conduct a sufficient safety assessment before using this method.

Reviewer: ```suggestion
> ⚠️ **Risk Warning**: With the privileged mode being enabled by `privileged: true`, the container has access to all devices of the host, that is, it has full control of the host's kernel. Improper uses will bring serious safety hazards. Please conduct a thorough safety assessment before using it.
```
